tools/codegen/src/generator/languages/rust/enumerator.rs

```diff
@@ -26,7 +26,7 @@ impl GenEnumerator for ast::Union {
             ref entity_union_item_paths,
             ref reader_union_item_paths,
         ) = {
-            self.items().iter().enumerate().fold(
+            self.items().iter().map(|item| (item.id(), item)).fold(
```

Comment:
suggest to keep iterator simple and modify the inner loop from `(index, inner)` to `item` and get the id by using `item.id()`
```suggestion
            self.items().iter().fold(
```